app/models/package_manager/go.rb

```diff
@@ -142,8 +143,26 @@ def self.mapping(raw_project)
       if raw_project[:html]
         url = raw_project[:overview_html]&.css(".UnitMeta-repo a")&.first&.attribute("href")&.value
 
+        # find an existing project with the same repository and replace the name with the existing project
+        # this will avoid creating duplicate Projects with various casing
+
+        # if this is a versioned module, make sure to find the right versioned project
+        regex_matches = raw_project[:name].match(VERSION_MODULE_REGEX)
+        if regex_matches
+          # try and find a versioned name matching this repository_url
+          existing_project_name = Project.where(platform: "Go").where("lower(repository_url) = :repo_url and name like :name", repo_url: url.downcase, name: "%/#{regex_matches[2]}").first&.name
```

Comment:
Was there an advantage to downcasing the repo url? I think pkg.go.dev should give us the canonical casing. 

One of my concerns with this whole thing too is that there's a git repository out there that might be case-sensitive and treat `repo.com/foo/bar` as a different repo than `repo.com/FOO/BAR`.